### PR TITLE
New stereo rendering mechanism

### PR DESCRIPTION
**Pull request by jherico**
This PR changes our stereo rendering approach by putting stereo functionality directly into the GPU backend.  Rather than running displaySide twice, we now run each gpu::Batch twice (unless it's specifically marked otherwise).  This drastically reduces the overhead for stereo modes and should move them from being 40-50% of mono displays to something like 80-90%.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9a2ad30f99043e1b2c5eecf1a589746700c9c315 available at: https://gist.github.com/60d21dace763ee2f0d7f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1105,57 +1104,54 @@ void Application::paintGL() {
...
-                // clear the normal and specular buffers
-                batch.clearFramebuffer(
-                    gpu::Framebuffer::BUFFER_COLOR0 |
-                    gpu::Framebuffer::BUFFER_COLOR1 |
-                    gpu::Framebuffer::BUFFER_COLOR2 |
-                    gpu::Framebuffer::BUFFER_DEPTH,
-                    vec4(vec3(0), 1), 1.0, 0.0);
-            });
-        }
+        doInBatch(&renderArgs, [&](gpu::Batch& batch) {
```
Comment (samcake) [inline interface/src/Application.cpp]: This batch is not needed anymore, it s done in one of the first job of the render pipeline

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1e074ab81c20cd2ef3bf60cb271988a238a9e6aa available at: https://gist.github.com/60d21dace763ee2f0d7f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1e074ab81c20cd2ef3bf60cb271988a238a9e6aa available at: https://gist.github.com/60d21dace763ee2f0d7f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.dmg

**Comment by hyperlogic** [comment]
So, when viewing this in the DK2 on the PC, I see the main scene jutter quite a bit, but the overlay rendering does not.  The PC is running at a pretty solid 75 htz.

**Comment by hyperlogic** [comment]
Also, when looking down at MyAvatar, it oddly doesn't seem stable, it jitters even when the HMD is stationary.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 72b1801b94a7751183758495a12055db45c72799 available at: https://gist.github.com/60d21dace763ee2f0d7f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.dmg

**Comment by jherico** [comment]
test this please

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 72b1801b94a7751183758495a12055db45c72799 available at: https://gist.github.com/60d21dace763ee2f0d7f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.dmg

**Comment by thoys** [comment]
That build of 26 minutes ago worked for me on Windows 10, without having the Oculus Runtime installed.

**Comment by hifi-gustavo** [comment]
Build console output for commit: cf3d84dbff8263fd0c9a50911417bf688da9b90c available at: https://gist.github.com/60d21dace763ee2f0d7f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5621.dmg

**Comment by judasshuffle** [comment]
This isn't right with the dk1. Left and right eye view is outta whack using sdk6.0.1

**Comment by ZappoMan** [comment]
Judas. Thanks for the note. We will look into it.

**Pull request merged**